src/components/post-list.astro

```diff
@@ -1,19 +1,24 @@
 ---
 import TagList from "./tag-list.astro";
-import Timestamp from "./timestamp";
+import Timestamp from "./timestamp/timestamp";
 
 const { posts } = Astro.props;
 ---
 
 {
     posts?.length ? (
         <ul class="postList">
-            {posts.map((post: { id: any; title: any; created_at: string; tags: string[]; }) => (
+            {posts.map((post: { id: any; title: any; created_at: string; tags: string[]; adult_content: boolean }) => (
                 <li class="postListing">
-                    <span class="title"><a href={`/posts/${post.id}`}>{post.title || 'Untitled'}</a></span>
-                    <div>by Anonymous</div>
-                    <div class="timestamp"><Timestamp client:only="preact" time={post.created_at} /></div>
-                    <TagList tags={post.tags} />
+                    <div class="left">
+                        <span class="title"><a href={`/posts/${post.id}`}>{post.title || 'Untitled'}</a></span>
+                        <div>by Anonymous</div>
+                        <TagList tags={post.tags} />
+                    </div>
+                    <div class="right">
+                        <Timestamp client:only="preact" time={post.created_at} />
+                        {post.adult_content && <div>This post is flagged as 18+</div>}
```

Comment:
we could possibly come up with a bit of a snazzier method of displaying this, maybe a little red "18+" box or something